drivers/comparator/comparator_nrf_lpcomp.c

```diff
@@ -0,0 +1,441 @@
+		*nrf = NRF_LPCOMP_EXT_REF_REF0;
+		break;
+
+	case COMP_NRF_LPCOMP_EXT_REF_REF1:
+		*nrf = NRF_LPCOMP_EXT_REF_REF1;
+		break;
+
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+static int shim_nrf_lpcomp_ref_to_nrf(enum comp_nrf_lpcomp_ref shim,
+				      nrf_lpcomp_ref_t *nrf)
+{
+	int ret = 0;
+
+	switch (shim) {
+	case COMP_NRF_LPCOMP_REF_SUPPLY_1_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_1_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_2_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_2_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_3_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_3_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_4_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_4_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_5_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_5_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_6_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_6_8;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_7_8:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_7_8;
+		break;
+
+#if LPCOMP_REFSEL_RESOLUTION
+	case COMP_NRF_LPCOMP_REF_SUPPLY_1_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_1_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_3_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_3_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_5_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_5_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_7_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_7_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_9_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_9_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_11_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_11_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_13_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_13_16;
+		break;
+
+	case COMP_NRF_LPCOMP_REF_SUPPLY_15_16:
+		*nrf = NRF_LPCOMP_REF_SUPPLY_15_16;
+		break;
+#endif
+
+	case COMP_NRF_LPCOMP_REF_EXT_REF:
+		*nrf = NRF_LPCOMP_REF_EXT_REF;
+		break;
+
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+#if NRF_LPCOMP_HAS_HYST
+static int shim_nrf_lpcomp_hyst_to_nrf(enum comp_nrf_lpcomp_hyst shim,
+				       nrf_lpcomp_hyst_t *nrf)
+{
+	int ret = 0;
+
+	switch (shim) {
+	case COMP_NRF_LPCOMP_HYST_NOHYST:
+		*nrf = NRF_LPCOMP_HYST_NOHYST;
+		break;
+
+	case COMP_NRF_LPCOMP_HYST_ENABLED:
+		*nrf = NRF_LPCOMP_HYST_ENABLED;
+		break;
+
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+#endif
+
+static void shim_nrf_lpcomp_reconfigure(void)
+{
+	(void)nrfx_lpcomp_reconfigure(&shim_nrf_lpcomp_data0.config);
+}
+
+static int shim_nrf_lpcomp_get_output(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	if (!shim_nrf_lpcomp_is_resumed()) {
+		return -EPERM;
+	}
+
+	return nrfx_lpcomp_sample();
+}
+
+static int shim_nrf_lpcomp_set_trigger(const struct device *dev,
+				       enum comparator_trigger trigger)
+{
+	shim_nrf_lpcomp_stop();
+
+	switch (trigger) {
+	case COMPARATOR_TRIGGER_NONE:
+		shim_nrf_lpcomp_data0.event_mask = 0;
+		shim_nrf_lpcomp_data0.config.detection = NRF_LPCOMP_DETECT_CROSS;
+		break;
+
+	case COMPARATOR_TRIGGER_RISING_EDGE:
+		shim_nrf_lpcomp_data0.event_mask = NRFX_LPCOMP_EVT_EN_UP_MASK;
```

Comment:
The masks here should be updated to `NRF_LPCOMP_INT_*_MASK` to reflect the latest changes made to https://github.com/zephyrproject-rtos/hal_nordic/pull/205.